### PR TITLE
Add style checker

### PR DESCRIPTION
**Pull request by dasconnor**
This PR adds the Vale and Markdownlint style checkers to the Avalanche-Docs repo. It sets a number of custom style rules and applies those rules to the entirety of the `docs/subnets` file tree. I resolved all errors in that directory, but not suggestions or warnings.

All new files edited by PRs will automatically trigger a Markdownlint review, but files must be added to Vale's configuration in `.github/workflows/ci.yml` to be reviewed. I also added instructions for running the tools locally and how to respond to their feedback.

### CODE REVIEW

**Review by yulin-dong — CHANGES_REQUESTED** [review]
I was able to reproduce the lint error at https://github.com/ava-labs/avalanche-docs/actions/runs/3614628016/jobs/6091190197 locally with `textlint`. However, the question is that if we need to specify how to install and run `textlint` in `style-checker-notes.md` or remove it from CI as specified in `.github/workflows/linter.yml`. https://github.com/github/super-linter runs `textlint` plus `markdownlint`.
```
> npx textlint  --rule terminology docs/overview/getting-started/avalanche-consensus.md

/Users/yulindong/workspace/src/github.com/ava-labs/avalanche-docs/docs/overview/getting-started/avalanche-consensus.md
   78:104  ✓ error  Incorrect usage of the term: “Ava”, use “AVA” instead  terminology
  143:286  ✓ error  Incorrect usage of the term: “Ava”, use “AVA” instead  terminology
  153:103  ✓ error  Incorrect usage of the term: “Ava”, use “AVA” instead  terminology
  183:328  ✓ error  Incorrect usage of the term: “Ava”, use “AVA” instead  terminology

✖ 4 problems (4 errors, 0 warnings)
✓ 4 fixable problems.
Try to run: $ textlint --fix [file]
```

File: .github/styles/Vocab/Products/accept.txt [hunk 1]
```diff
@@ -50,9 +53,11 @@ EOAs
 Ethereum
 feeRecipient
 GitHub
+github
```
[the remark this answers is not in the document]
Comment (yulin-dong) [reply .github/styles/Vocab/Products/accept.txt:56]: I think I got errors of github on pathes. However, as long as we can get errors cleared, I am fine with just GitHub.

File: docs/overview/getting-started/intro.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 slug: /
 ---
+<!-- markdownlint-disable MD013 -->
```
[the remark this answers is not in the document]
Comment (yulin-dong) [reply docs/overview/getting-started/intro.md:4]: I got a lot of highlights on my VC editor without this line.

[the remark this answers is not in the document]
Comment (dasconnor) [reply docs/overview/getting-started/intro.md]: fixed

File: .github/workflows/linter.yml [hunk 3]
```diff
@@ -52,3 +52,4 @@ jobs:
           VALIDATE_JSCPD: false
           VALIDATE_YAML: false
           VALIDATE_GITLEAKS: false
+          VALIDATE_NATURAL_LANGUAGE: false
```
Comment (yulin-dong) [inline .github/workflows/linter.yml:55]: Cool trick!

**Review by yulin-dong — APPROVED** [review]

**Pull request merged**